app/src/main/java/com/example/habitsmasher/HabitList.java

```diff
@@ -65,7 +65,7 @@ public void setSnapshots(ObservableSnapshotArray<Habit> snapshots) {
      * @param date date of added habit
      * @param username user of the habit list habit is being added to
      */
-    public void addHabitToDatabase(String title, String reason, Date date, String username) {
+    public void addHabitToDatabase(String title, String reason, Date date, DaysTracker tracker, String username) {
```

Comment:
When adding/editing habits, the tracker is passed instead of the string of days so that when it comes down to actually sending the data to the DB, the string is from the source and editable until it is sent.